Geo grid aggregation circuit breaker tests can fail assertion

It looks like the search fails as expected, but the failures aren't in the right format. In the test run below, only 1 out of 2 shards fails, so we don't return the expected status code/ exception.

Example reproduce line:
```
./gradlew ':x-pack:plugin:spatial:integTestRunner' --tests "org.elasticsearch.xpack.spatial.SpatialClientYamlTestSuiteIT.test {yaml=/30_geotile_grid/Test geotile_grid aggregation circuit breaker}" \
  -Dtests.seed=89B477E68892C5AA \
  -Dtests.security.manager=true \
  -Dtests.locale=ga-IE \
  -Dtests.timezone=Asia/Baku \
  -Dcompiler.java=14 \
  -Druntime.java=8
```

Build scan: https://gradle-enterprise.elastic.co/s/blopfoqljmy2k

Relevant excerpt from logs:
```
Caused by: |  
-- | --
  | java.lang.AssertionError: expected [4xx\|5xx] status code but api [search] returned [200 OK] [{"took":11,"timed_out":false,"_shards":{"total":2,"successful":1,"skipped":0,"failed":1,"failures":[{"shard":1,"index":"locations","node":"JyLzhPoqS22zFQMXdGvbag","reason":{"type":"circuit_breaking_exception","reason":"[request] Data too large, data for [<agg [grid]>] would be [26344/25.7kb], which is larger than the limit of [25600/25kb]","bytes_wanted":26344,"bytes_limit":25600,"durability":"TRANSIENT"}}]},"hits":{"total":0,"max_score":null,"hits":[]},"aggregations":{"grid":{"buckets":[]}}}] |  
  | at org.junit.Assert.fail(Assert.java:88) |  
  | at org.elasticsearch.test.rest.yaml.section.DoSection.execute(DoSection.java:291) |  
  | at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:398) |  
  | ... 38 more
```

--- a/x-pack/plugin/spatial/src/test/resources/rest-api-spec/test/30_geotile_grid.yml
+++ b/x-pack/plugin/spatial/src/test/resources/rest-api-spec/test/30_geotile_grid.yml
@@ -58,13 +58,13 @@
 
 ---
 "Test geotile_grid aggregation circuit breaker":
-  - skip:
-      version: "all"
-      reason: "AwaitsFix https://github.com/elastic/elasticsearch/issues/56257"
   - do:
       indices.create:
         index: locations
         body:
+          settings:
+            number_of_shards: 1
+            number_of_replicas: 0
           mappings:
             properties:
               location:

